### DISCUSSION TITLE
Get access to GraphQL request query

### DISCUSSION THREAD

**Discussion by nagg-hash** [ChilliCream/hotchocolate#5323] [Q&A] [answered]
Currently I am using GraphQl(HotChocolate) in a .net 6 project with Cosmos DB(SQL API).
I was just looking for a way to access GraphQL request query and transform it into a SQL query and use this to fetch data from Cosmos container(Idea here is to fetch only selected field from Cosmos container).

Any easy way to access GraphQL Request query? or 
any other better solution for above use case?

```
public class Query
{
    public async Task<IEnumerable<Data>> GetData([Service] IDataService service)
    {
        //get grapghQl
       //transform to sql query
        var _data = await service.GetContainerData(query);
        return _data ;
    }
}

**Comment by tobias-tengler** [comment, marked as answer]
If you are using Entity Framework you can use the projection integration:
https://chillicream.com/docs/hotchocolate/fetching-data/projections
